### ISSUE TITLE
Plugin override has unexpected edge cases

### ISSUE THREAD

**Issue by jen20** [hashicorp/terraform#6670]
### Terraform Version
`master` (0.7-dev)

### Affected Resource(s)
Please list the resources as a list, for example:
- null_resource
- any other resource in any other providers

Compile Terraform using `make dev`, and then separately compile the AWS plugin using `make plugin-dev PLUGIN=provider-aws`.

### Terraform Configuration Files
```hcl
provider "aws" {}
provider "null" {}

resource "aws_vpc" "test" {
    cidr_block = "10.0.0.0/24"
}

resource "null_resource" "noop" {}
```

### Debug Output
```
[WARN] /Users/James/Code/go/bin/terraform-provider-aws overrides an internal plugin for aws-provider.
  If you did not expect to see this message you will need to remove the old plugin.
  See https://www.terraform.io/docs/internals/internal-plugins.html
provider.aws.region
  The region where AWS operations will take place. Examples
  are us-east-1, us-west-2, etc.

  Default: us-east-1
  Enter a value:

Error configuring: 1 error(s) occurred:

* Unrecognized remote plugin message: [WARN] /Users/James/Code/go/bin/terraform-provider-aws overrides an internal plugin for aws-provider.

This usually means that the plugin is either invalid or simply
needs to be recompiled to support the latest protocol.
```

### Expected Behavior

The overridden plugin is used, this message does not appear. Compiling the null provider makes the same configuration work with the same AWS plugin so it is possible there is some error being interpreted incorrectly in the plugin loading code.

**Comment by catsby** [comment]
I observed this early on, but have since not been able to reproduce it.

Steps:

- remove all `terraform*` from `$GOPATH/bin`
- `make dev`
- `make plugin-dev PLUGIN=provider-aws`

I still get the override warning , but I don't get the remote plugin error

**Comment by Bowbaq** [comment]
I'm running into this, working on the the `azurerm` provider & using a `template_file` resource in my configuration.

Steps:
```bash
rm -f $GOPATH/bin/terraform*

cd $GOPATH/src/github.com/hashicorp/terraform
make dev
make plugin-dev PLUGIN=provider-azurerm

TMP=$(mktemp -d)
cat > $TMP/main.tf <<EOF
resource "azurerm_resource_group" "test" {
  name     = "test"
  location = "West US"
}

resource "template_file" "test" {
  template = "Hi"
}
EOF
(cd $TMP && terraform plan)
cd - && make plugin-dev PLUGIN=provider-template
(cd $TMP && terraform plan)
```

**Comment by stack72** [comment]
@jen20 I believe this has been fixed already by @cbednarski 

I can no longer reproduce this

**Comment by cbednarski** [comment]
Oh yeah. The fix was to prevent the plugin from emitting the log message during startup, which caused the RPC layer to be unhappy. I was notified about this via chat and didn't realize there was a GitHub issue for this.

**Comment by cbednarski** [comment]
I think this was fixed in #6906. Please @ me if you are still running into this.

**Comment by stack72** [comment]
@cbednarski thanks for commenting to let me know :)
